lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -111,6 +113,27 @@ class PolicyElementAssociation < ::ActiveRecord::Base
       has_and_belongs_to_many :operations, class_name: :"PolicyMachineStorageAdapter::ActiveRecord::Operation"
       belongs_to :user_attribute
       belongs_to :object_attribute
+
+      #TODO: ActiveRecord's generated operations= method is inefficient, makes 1 query for each op added or removed even though there's no hooks
+      # Awkward manual implementation for now, but in the future change this to an hstore or something in the postgres adapter,
+      # and/or fix Rails.
+      def operations=(updated_operations)
+        updated_operation_set = Set.new(updated_operations)
+        current_operation_set = Set.new(self.operations)
+        new_operations = updated_operation_set - current_operation_set
+        removed_operations = current_operation_set - updated_operation_set
+        transaction do
+          OperationsPolicyElementAssociation.where(policy_element_association_id: self.id)
+                                            .where(operation_id: removed_operations.map(&:id))
+                                            .delete_all
+          OperationsPolicyElementAssociation.import([:policy_element_association_id, :operation_id],
+                                                     new_operations.map{ |op| [self.id, op.id] },
+                                                     validate: false)
```

Comment:
There are no validations, I was just worried the gem would instantiate model elements if I didn't pass that in.